### PR TITLE
Fix solarized themes colors

### PR DESCRIPTION
**Pull request by ericbn**
- Use colors according to
  http://ethanschoonover.com/solarized#usage-development, namely dark
  mode core colors are only base1, base0, base01, base03, base03
- 'altercation/vim-colors-solarized' also uses base00 in dark mode for
  the VertSplit highlight, so use this too (has more contrast)
- Use colors as to optimize contrast, and as to make the same scheme
  work for light mode by just inverting the colors (base03 becomes base3
  and so on, as in 'altercation/vim-colors-solarized')
- Refactor removing duplicate code

### CODE REVIEW

**Comment by ericbn** [comment]
## Dark
### Before
screen shot 2016-08-07 at 10 27 00 am
Base3 and base2 are not part of the dark mode, not optimal contrast in other parts (most notably the base1/base01 foreground/background pair).
### After
screen shot 2016-08-07 at 11 45 12 am
## Light
### Before
screen shot 2016-08-07 at 11 03 43 am
Base03 and base02 are not part of the light mode, not optimal contrast in other parts (most notably the base01/base1 foreground/background pair).
### After
screen shot 2016-08-07 at 11 47 09 am

File: autoload/lightline/colorscheme/solarized.vim [hunk 1]
```diff
@@ -2,10 +2,87 @@
...
+  let s:temp01 = s:base01
+  let s:temp00 = s:base00
+  let s:base03 = s:base3
+  let s:base02 = s:base2
+  let s:base01 = s:base1
+  let s:base00 = s:base0
+  let s:base0 = s:temp00
+  let s:base1 = s:temp01
+  let s:base2 = s:temp02
+  let s:base3 = s:temp03
```
Comment (itchyny) [inline autoload/lightline/colorscheme/solarized.vim]: We do not need temporary variables to swap.
```vim
  let [s:base03, s:base3] = [s:base3, s:base03]
```

Comment (ericbn) [inline autoload/lightline/colorscheme/solarized.vim]: Cool. Fixed this now.

**Comment by itchyny** [comment]
Thank you.

**Pull request merged**